### ISSUE TITLE
[Outlook] The mailbox.item.to.getAsync property returns no value when add-in is opened for the first time

### ISSUE THREAD

**Issue by splinter** [OfficeDev/office-js#486]
The to addresses list is empty when the following snippet of code is executed the first time
the add-in is loaded in compose mode for a mail item [1]:

```js
  Office.context.mailbox.item.to.getAsync(callback);
```

This behaviour seems to be intermittent.I have not been able to identify any pattern for the conditions which cause this issue.

## Expected Behavior
The to address list should be always populated with addresses entered in the mail item compose form.

## Current Behavior
The to address list is empty during the first invocation intermittently. 

## Steps to Reproduce, or Live Example
1. Open the compose mail item view
2. Enter an email address in the to field
3. Open an Outlook add-in
4. Attempt to fetch to addresses with [1]

## Context
The users are not able to interact with the to addresses within the add-in unless they close
and open the add-in again or manually initiate an action to reload the addresses list.

## Your Environment
* Platform [PC desktop, Mac, iOS, Office Online]: Windows
* Host [Excel, Word, PowerPoint, etc.]: Outlook 2016
* Office version number: 1903 (Build 11425.20228)
* Operating System: Windows 10

## References
[1] https://docs.microsoft.com/en-us/office/dev/add-ins/reference/objectmodel/requirement-set-1.5/office.context.mailbox.item?view=office-js#compose-mode-7

**Comment by exextoc** [comment]
This API will only return **Resolved** Recipients. If an E-mail Address hasn't been resolved then the API will not return those e-mails until they are resolved. If you want to grab e-mails after you have queried for to.getAsync(), the RecipientsChanged Event may be useful (https://docs.microsoft.com/en-us/office/dev/add-ins/reference/objectmodel/requirement-set-1.7/office#eventtype-string) It should fire when a recipient is newly resolved. 

Are you by any chance using the Pinning Feature and not registering for the ItemChanged Event? 

Otherwise, if the API doesn't return resolved recipients, we haven't been able to repro this and this isn't a known issue. Any additional information you may have via the Repro Steps would be helpful.

**Comment by exextoc** [comment]
Are you by any chance using the Pinning Feature and not registering for the ItemChanged Event?

**Comment by splinter** [comment]
We are using the Pinning feature but we are registering for the ItemChanged event.

**Comment by emtencza** [comment]
Are the recipients resolved when making the mailbox.item.to.getAsync API call? If not, the API will not return those recipients until they are resolved, which may be why you are seeing an empty address list.

**Comment by exextoc** [comment]
Are the recipients resolved when you call mailbox.item.to.getAsync? If not, the API will not return the recipients until they are resolved, which may be the reason you are seeing an empty address list.

**Comment by splinter** [comment]
We are retrieving the addresses based on mailbox.item.to.getAsync  in compose mode and mailbox.item.to in read mode as per the documentation [1].

[1] https://docs.microsoft.com/en-us/office/dev/add-ins/reference/objectmodel/requirement-set-1.5/office.context.mailbox.item?view=office-js#to-arrayemailaddressdetailsrecipients

**Comment by exextoc** [comment]
If the recipient are **not resolved**. They will not be returned. (sorry to repeat this, but you never answered whether or not the recipients were resolved in the previous replies) As of today that is the only known case where stuff in the recipient well is not returned. (and by design). 

Otherwise, we haven't been able to repro this, so any additional information (videos, screenshots, repro steps) will help. Are you polling for recipients? or is there are button you push to get the recpients?

**Comment by splinter** [comment]
Can you please elaborate by what you mean by "resolved"? If by resolved you mean detecting their display names then yes.

**Comment by exextoc** [comment]
Here is an example in OWA but the same behavior exists for the API on Outlook Desktop too:
![image](https://user-images.githubusercontent.com/28549921/62561459-20d2b200-b834-11e9-85f0-6213b1a74d35.png)
Jesse is a resolved recipient, Kat is an unresolved recipient. Only resolved recipients will be returned from the Recipients API. 

Can you share more about particular scenarios where the API is not updated? Are you polling for recipients or trying to listen for when the recipients well changes?

**Comment by kbrandl** [comment]
@splinter do you need more assistance with this issue, or has the feedback provided by @exextoc unblocked you? Thanks!

**Comment by rgmills** [comment]
Hi, I have a question regarding this behavior. (Please let me know if we should move to another issue).

In the case of calling `Office.context.mailbox.item.to.getAsync(callback);` when there are only contacts who are not resolved, Outlook is removing those contacts from the To recipient list.

To reproduce:

- Open new email message
- Manually type in an smtp email address, hit esc to switch focus
- Invoke `Office.context.mailbox.item.to.getAsync(callback);` in an add-in
- Observe the smtp email address has been removed

Outlook Info:
- Version 1908
- Build 16.0.11929.20288

I've checked in the Outlook Web client and it does not remove the unresolved smtp email address.

**Comment by exextoc** [comment]
Hi @rgmills, please open another issue as your issue is different from this one, and we'll be sure to take a look at it asap.

**Comment by rgmills** [comment]
@exextoc Opened https://github.com/OfficeDev/office-js/issues/738
